code/modules/antagonists/eldritch_cult/eldritch_magic.dm

```diff
@@ -0,0 +1,458 @@
+	. = ..()
+	playsound(user, 'sound/items/welder.ogg', 75, TRUE)
+	if(ishuman(target))
+		var/mob/living/carbon/human/tar = target
+		if(tar.anti_magic_check())
+			tar.visible_message("<span class='danger'>Spell bounces off of [target]!</span>","<span class='danger'>The spell bounces off of you!</span>")
+			return
+	var/datum/mind/M = user.mind
+	var/datum/antagonist/e_cult/cultie = M.has_antag_datum(/datum/antagonist/e_cult)
+
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		C.adjustBruteLoss(10)
+		C.AdjustKnockdown(5 SECONDS)
+		C.adjustStaminaLoss(60)
+
+	for(var/X in cultie.get_all_knowledge())
+		var/datum/eldritch_knowledge/EK = X
+		EK.on_mansus_grasp(target, user, proximity_flag, click_parameters)
+
+/obj/effect/proc_holder/spell/aoe_turf/rust_conversion
+	name = "Aggressive Spread"
+	desc = "Spreads rust onto nearby turfs."
+	school = "transmutation"
+	charge_max = 300 //twice as long as mansus grasp
+	clothes_req = FALSE
+	invocation = "A'GRSV SPR'D"
+	invocation_type = "whisper"
+	range = 3
+	action_icon = 'icons/mob/actions/actions_ecult.dmi'
+	action_icon_state = "corrode"
+	action_background_icon_state = "bg_ecult"
+
+/obj/effect/proc_holder/spell/aoe_turf/rust_conversion/cast(list/targets, mob/user = usr)
+	playsound(user, 'sound/items/welder.ogg', 75, TRUE)
+	for(var/turf/T in targets)
+		///What we want is the 3 tiles around the user and the tile under him to be rusted, so min(dist,1)-1 causes us to get 0 for these tiles, rest of the tiles are based on chance
+		var/chance = 100 - (max(get_dist(T,user),1)-1)*100/(range+1)
+		if(!prob(chance))
+			continue
+		T.rust_heretic_act()
+
+/obj/effect/proc_holder/spell/aoe_turf/rust_conversion/small
+	name = "Rust Conversion"
+	desc = "Spreads rust onto nearby turfs."
+	range = 2
+
+/obj/effect/proc_holder/spell/targeted/touch/ash_leech
+	name = "Blood Siphon"
+	desc = "Touch spell that heals you while damaging the enemy."
+	hand_path = /obj/item/melee/touch_attack/ash_leech
+	school = "evocation"
+	charge_max = 150
+	clothes_req = FALSE
+	invocation = "FL'MS O'ET'RN'ITY"
+	invocation_type = "whisper"
+	action_icon = 'icons/mob/actions/actions_ecult.dmi'
+	action_icon_state = "blood_siphon"
+	action_background_icon_state = "bg_ecult"
+
+/obj/item/melee/touch_attack/ash_leech
+	name = "Blood Siphon"
+	desc = "A sinister looking aura that distorts the flow of reality around it."
+	icon_state = "disintegrate"
+	inhand_icon_state = "disintegrate"
+	catchphrase = "R'BRTH"
+
+/obj/item/melee/touch_attack/ash_leech/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	playsound(user, 'sound/magic/demon_attack1.ogg', 75, TRUE)
+	if(ishuman(target))
+		var/mob/living/carbon/human/tar = target
+		if(tar.anti_magic_check())
+			tar.visible_message("<span class='danger'>Spell bounces off of [target]!</span>","<span class='danger'>The spell bounces off of you!</span>")
+			return
+	var/mob/living/carbon/C2 = user
+	if(isliving(target))
```

Comment:
You should have a check for whether or not the target is dead or alive and cause an early return. Otherwise you can use this to regenerate off corpses.

Reply:
healing off of corpses isn't really a big deal since it is on 30 seconds cooldown and you can fully heal yourself with the brute kit. I dont mind this.